setuptools/extension.py

```diff
@@ -126,10 +127,12 @@ class Extension(_Extension):
       specified on Windows. (since v63)
     """
 
-    def __init__(self, name, sources, *args, **kw):
+    def __init__(
+        self, name: str, sources: list[str], *args, py_limited_api: bool = False, **kw
```

Comment:
Jason might have added support for `PathLike`sources recently in https://github.com/pypa/distutils/pull/237 (https://github.com/pypa/setuptools/blob/v72.2.0/setuptools/_distutils/extension.py#L29).

(PS.: granted they are incompatible with `SETUPTOOLS_USE_DISTUTILS=stdlib` but that is a deprecated code path).

Reply:
Thanks for the heads up, that added annotation isn't necessary for this PR. I'd rather avoid possible type conflicts with sdtlib distutils for now. I'll tackle that later.